internal/timewarp/timewarp.go

```diff
@@ -295,7 +295,7 @@ func timeWarpPyPIProjectRequest(client httpx.BasicClient, obj map[string]any, at
 				if t.Before(at.Add(time.Second)) {
 					pastFiles = append(pastFiles, file)
 				}
-				if t.Before(firstSeen) {
+				if t.Before(firstSeen) || firstSeen.IsZero() {
```

Comment:
I don't seem to see where this is being tested?

Reply:
it is tested in the version part of the URL that gets requested

Reply:
to clarify: firstSeen will never update in the loop if this isn't here so the block below was using the default behavior of omitting latestVersion (which, for our purposes, has the same effect of fetching the actual latest version). added another test to exercise that default path, as well.